docs: basic view components documentation (#1891)
pankod/refine: documentation/docs/ui-frameworks/mui/components/basic-views/edit.md

---
id: edit
title: Edit
---

import actionButtonsUsage from '@site/static/img/guides-and-concepts/basic-views/edit/actionButtonUsageMui.png'
import cardHeaderPropsTitle from '@site/static/img/guides-and-concepts/basic-views/edit/cardHeaderPropsTitle.png'

`<Edit>` provides us a layout for displaying the page. It does not contain any logic but adds extra functionalities like a refresh button.

We will show what `<Edit>` does using properties with examples.

## Properties

### `canDelete` and `deleteButtonProps`

> `canDelete` allows us to add the delete button inside the `<Edit>` component. If the resource has the `canDelete` property,refine adds the delete button by default. If you want to customize this button you can use the `deleteButtonProps` property like the code below.

Reviewer: ```suggestion
`canDelete` allows us to add the delete button inside the `<Edit>` component. If the resource has the `canDelete` property, refine adds the delete button by default. If you want to customize this button you can use the `deleteButtonProps` property like the code below.
```

```tsx
import { Edit, Button } from "@pankod/refine-mui";

export const EditPage: React.FC = () => {
    return (
        <Edit
            cardHeaderProps={{
                title: <Typography variant="h5">Custom Title</Typography>,
            }}
        >
            ...
        </Edit>
    );
};
```

<br/>
<div class="img-container">
    <div class="window">
        <div class="control red"></div>
        <div class="control orange"></div>
        <div class="control green"></div>
    </div>
    <img src={cardHeaderPropsTitle} alt="actionButton Usage" />
</div>
<br/>

### `cardContentProps`

`<Edit>` uses the Material UI `<CardContent>` components so you can customize with the props of `cardContentProps`.

### `cardActionsProps`

`<Edit>` uses the Material UI `<CardActions>` components so you can customize with the props of `cardActionsProps`.

### `recordItemId`

The `<Edit>` component reads the `id` information from the route by default. `recordItemId` is used when it cannot read from the URL(when used on a custom page, modal or drawer).

```tsx
import { useModalForm } from "@pankod/refine-react-hook-form";

> import { Drawer, ...} from "@pankod/refine-mui";

Reviewer: ```suggestion
import { Drawer, Edit } from "@pankod/refine-mui";
```

---
id: edit
title: Edit
---

import actionButtonsUsage from '@site/static/img/guides-and-concepts/basic-views/edit/actionButtonUsageMui.png'
import cardHeaderPropsTitle from '@site/static/img/guides-and-concepts/basic-views/edit/cardHeaderPropsTitle.png'

> `<Edit>` provides us a layout for displaying the page. It does not contain any logic but adds extra functionalities like a refresh button.

Reviewer 2: ```suggestion
`<Edit>` provides us a layout for displaying the page. It does not contain any logic but adds extra functionalities like a `<RefreshButton />`.
```

can be add link

| ----------------- | ------------------------------------------------------------------------------- | ------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
| canDelete         | Adds a delete button                                                            | `boolean`                                                                 | If the resource has `canDelete` prop it is `true` else `false` `false`         |
| deleteButtonProps | Adds properties for delete button                                               | [`DeleteButtonProps`](/core/interfaces.md#delete-button-props)            | `<DeleteButton>`                                                               |
| saveButtonProps   | Adds props for edit button                                                      | `{ disabled: boolean; onClick: () => void; loading: boolean; }`           | `<SaveButton>`                                                                 |

> | actionButtons     | Passes the props for `<CardActions>`                                            | `React.ReactNode`                                                         | `<SaveButton>` and depending on your resource configuration (`canDelete` prop) |

Reviewer 2: .
